src/common/core/users/metarules.js

```diff
@@ -0,0 +1,311 @@
+/*globals define*/
+/*jshint node:true, browser:true, newcap:false*/
+/**
+ * @author kecso / https://github.com/kecso
+ * @author pmeijer / https://github.com/pmeijer
+ */
+
+define(['q'], function (Q) {
+    'use strict';
+
+    // Helper functions.
+    function loadNode(core, rootNode, nodePath) {
+        var deferred = new Q.defer();
+
+        core.loadByPath(rootNode, nodePath, function (err, node) {
+            if (err) {
+                deferred.reject(new Error(err));
+            } else if (core.isEmpty(node)) {
+                deferred.reject(new Error('Given nodePath does not exist "' + nodePath + '"!'));
+            } else {
+                deferred.resolve(node);
+            }
+        });
+
+        return deferred.promise;
+    }
+
+    function loadNodes(core, node, nodePaths) {
```

Comment:
wouldn't it be enough to use Q.nfcall(core.loadByPath...), and then there is no need for the loadNode function?